dll/win32/shell32/shfldr.h

```diff
@@ -27,11 +27,24 @@
 #define CHARS_IN_GUID 39
 
 typedef struct {
-    int colnameid;
-    int pcsFlags;
-    int fmt;
-    int cxChar;
+    WORD colnameid; // Column title text resource id passed to LoadString
+    WORD colstate;  // SHCOLSTATEF
```

Comment:
Isn't it possible to directly use the SHCOLSTATEF type here, instead of "WORD" ?

Reply:
> instead of "WORD"

No, don't do that. I packed them into two words on purpose.

Reply:
On purpose? why? Does the data get stored in the registry with a format compatible with Windows' shell?

Reply:
> why?

Because the values fit and will save `sizeof(int)` of constant data in all columns in all of the folder implementations.

Reply:
yeahok if you can ensure that all values of colstate cannot be larger than 0xFFFF (and same for column IDs), I may understand. But that's the kind of "micro-optimization" that doesn't make that much sense in general.

Reply:
> column IDs

Technically the column string/title resource id but yes, both the resource id (`IS_INTRESOURCE`) and the actual column ID (`SHCIDS_COLUMNMASK`)  are restricted to WORD.

The relevant SHCOLSTATEF flags also fit in a WORD.

Reply:
Ok. Maybe make this code comment more explicit than bare 'SHCOLSTATEF'?

Reply:
> more explicit than bare 'SHCOLSTATEF'?

It could be changed to something like `// SHCOLSTATEF returned by IShellFolder2::GetDefaultColumnState` if that makes it clearer.

Reply:
Maybe, though I am thinking more about what you explained about using `WORD` here.

Reply:
I agree with @HBelusca here, let's not introduce difficult to understand optimizations unless they are strictly necessary.

Reply:
@cbialorucki, I do not understand what you are suggesting. Can you please provide code suggestions and be more specific about what you mean? Thanks.